docs: document GitHub App credentials and manifest flow (#1302)
seqeralabs/docs: platform-cloud/docs/git/overview.md

1. Enter your **Username** and **Access token**.
1. (Recommended) Enter the **Repository base URL** for which the credentials should be applied. This option is used to apply the provided credentials to a specific repository, e.g., `https://github.com/seqeralabs`.

### GitHub App

> As an alternative to personal access tokens, you can authenticate Seqera Platform to GitHub using a [GitHub App](https://docs.github.com/en/apps/creating-github-apps/about-creating-github-apps/about-creating-github-apps). GitHub Apps are the GitHub-recommended way to integrate with the GitHub API: they act on their own behalf rather than impersonating a user, support fine-grained permissions scoped to specific repositories, and use short-lived installation tokens that are not tied to a single account.

Reviewer: ```suggestion
Authenticate Seqera Platform to GitHub using [GitHub Apps](https://docs.github.com/en/apps/creating-github-apps/about-creating-github-apps/about-creating-github-apps), the GitHub-recommended way to integrate with the GitHub API. They act on their own behalf rather than impersonating a user, support fine-grained permissions that target specific repositories, and use short-lived installation tokens that don't belong to a single account.
```

When you select **GitHub** as the **Provider**, the credentials form shows a **GitHub credential type** selector with two tabs:

> - **Access token** — Authenticate using a personal access token (PAT) for API access. This is the legacy flow described in the [GitHub](#github) section above.

Reviewer: ```suggestion
- **Access token**: Authenticate using a personal access token (PAT) for API access. See the legacy flow in the [GitHub](#github) section above.
```

> - **GitHub App** — Set up app-based authentication with dedicated credentials. When you select this tab, a second selector lets you choose between two flows:

Reviewer: ```suggestion
- **GitHub App**: Set up app-based authentication with dedicated credentials. Select this tab to access a second selector with two flows:
```

>   - **Create and add** — Use the GitHub App manifest flow to create a new app on GitHub directly from Seqera. Seqera generates a pre-filled manifest, redirects you to GitHub for approval, then automatically retrieves and stores the resulting App ID, private key, client secret, and webhook secret.

Reviewer: ```suggestion
  - **Create and add**: Use the GitHub App manifest flow to create a new app on GitHub directly from Seqera. Seqera generates a pre-filled manifest, redirects you to GitHub for approval, then automatically retrieves and stores the resulting App ID, private key, client secret, and webhook secret.
```

>   - **Add preexisting** — Register an app you have already created on GitHub by entering its App ID, installation ID, private key, and other security keys manually.

Reviewer: ```suggestion
  - **Add preexisting**: Register an app you have already created on GitHub by entering its App ID, installation ID, private key, and other security keys manually.
```

> The manifest flow (**Create and add**) is recommended for new integrations: it eliminates the manual copy-paste of multiple secrets, ensures the app is created with the minimum required permissions (`contents: read`, `metadata: read`), and avoids configuration errors. Use **Add preexisting** only when the app already exists or when you must create the app outside of Seqera.

Reviewer: ```suggestion
For new integrations, use the manifest flow (**Create and add**). It removes the manual copy-paste of secrets, ensures the app is created with the minimum required permissions (`contents: read`, `metadata: read`), and avoids configuration errors. Use **Add preexisting** only if the app already exists or must be created outside of Seqera.
```

![GitHub App credentials form showing the Access token / GitHub App tabs and the Create and add / Add preexisting sub-selector](./_images/credentials-github-app-form.png)

#### Create a new GitHub App from Seqera

> To create and install a GitHub App from Seqera using the manifest flow:

Reviewer: ```suggestion
To create and install a GitHub App from Seqera with the manifest flow:
```

> 1. From an organization workspace: Select **Credentials** > **Add Credentials**. From your personal workspace: Go to the user menu and select **Your credentials** > **Add credentials**.

Reviewer: ```suggestion
1. Go to the credentials page:
    - Organization workspace: Select **Credentials** > **Add Credentials**.
    - Personal workspace: Select your user menu, then select **Your credentials** > **Add credentials**.

```

> 1. Enter a **Name** for the new credentials, e.g., `my-github-app`. Underscores in the credential name are replaced with spaces in the resulting GitHub App name (`Seqera Platform - my github app`).

Reviewer: ```suggestion
1. Enter a **Name** for the new credentials, for example, `my-github-app`.

    :::note
    Underscores in the credential name are replaced with spaces in the resulting GitHub App name  (e.g., `Seqera Platform - my github app`).
    ::::
```

1. Select **GitHub** as the **Provider**, set the **GitHub credential type** to **GitHub App**, then select **Create and add**.
1. Enter the **GitHub URL**:
    - For GitHub.com, leave the default value (`https://github.com`).

>     - For a GitHub Enterprise Server instance, enter the base URL of your instance (for example, `https://github.example.com`). HTTPS is required, and private or loopback addresses are rejected.

Reviewer: ```suggestion
    - For a GitHub Enterprise Server instance, enter the base URL of your instance (e.g., `https://github.example.com`). HTTPS is required. Private or loopback addresses are rejected.
```

> 1. (Optional) Enter the **GitHub repository URL** to scope access to a single repository, e.g., `https://github.com/seqeralabs/nf-tower`. Leave this field empty to create credentials that are not bound to a specific repository.

Reviewer: ```suggestion
1. (Optional) Enter the **GitHub repository URL** to scope access to a single repository, for example, `https://github.com/seqeralabs/nf-tower`. Leave this field empty to create credentials that are not bound to a specific repository.
```

1. Select the **App scope**:
    - **Organization** — App owned by an organization (requires admin access). Enter the **GitHub organization name** (case-sensitive). You must be an **owner** of the target organization to create an app on its behalf.

>     - **Personal** — App owned by your personal GitHub account. The **GitHub organization name** field is hidden.

Reviewer: ```suggestion
    - **Organization**: App owned by an organization (requires admin access). Enter the **GitHub organization name** (case-sensitive). You must be an **owner** of the target organization to create an app on its behalf.
    - **Personal**: App owned by your personal GitHub account. The **GitHub organization name** field is hidden.
```

1. Select **Create app on GitHub**. Seqera redirects you to GitHub:
    - For personal scope: `https://github.com/settings/apps/new`
    - For organization scope: `https://github.com/organizations/<your-org>/settings/apps/new`
    - For GitHub Enterprise Server, the equivalent path on your instance.

   The manifest is pre-filled with the app name, callback URL, webhook URL, and the required permissions (`contents: read`, `metadata: read`).

   ![GitHub "Create GitHub App" page with the manifest pre-filled, showing the app name "Seqera Platform - new github app"](./_images/credentials-github-mainfest-page.png)

1. On GitHub, review the requested permissions and select **Create GitHub App**. GitHub redirects you back to Seqera, which exchanges the temporary code for the app credentials and stores them in your workspace or personal credentials.
1. After the redirect, install the app on the repositories you want Seqera to access:
    - Open the new app on GitHub: **Settings** > **Developer settings** > **GitHub Apps** > **[your app]** > **Install App**.

>     - For an organization-owned app, select the organization. For a personal app, select your user account.

Reviewer: ```suggestion
    - For an organization-owned app, select the organization.
    - For a personal app, select your user account.
```

>     - Choose **Only select repositories** and add the specific repositories Seqera should access, or **All repositories** to grant access to all current and future repositories.

Reviewer: ```suggestion
    - Choose **Only select repositories** and add the specific repositories Seqera should access, or select **All repositories** to grant access to all current and future repositories.
```

    - Select **Install** to complete installation.

   ![GitHub App installation page showing "Only select repositories" with one or more repositories selected](./_images/credentials-github-install-app.png)

> The new credential appears in the **Credentials** list with the GitHub App icon. Credentials created from a workspace credentials page are scoped to that workspace; credentials created from your personal credentials page are scoped to your user and are not visible to any workspace.

Reviewer: ```suggestion
The new credential appears in your **Credentials** list with the GitHub App icon. Credentials created from your workspace credentials page are scoped to that workspace; credentials created from your personal credentials page are scoped to your user and are not visible to any workspace.
```

> :::info

Reviewer: ```suggestion
:::note
```

> If you cancel the manifest flow on GitHub or close the browser tab before approving the app, no credential is created on the Seqera side. The temporary state that protects the redirect against CSRF expires after 10 minutes and cannot be reused — restart the flow from the credentials form.

Reviewer: ```suggestion
If you cancel the manifest flow on GitHub or close the browser tab before approving the app, no credentials are created on Seqera. The temporary state that protects the redirect against CSRF expires after 10 minutes and cannot be reused. To try again, restart the flow from the credentials form.
```

:::

#### Add an existing GitHub App

> If you have already created and installed a GitHub App, register it in Seqera by setting the **GitHub credential type** to **GitHub App** and selecting **Add preexisting**, then entering the app's security keys (App ID, installation ID, app slug, private key, client secret, and webhook secret) along with the same **GitHub URL**, **App scope**, and optional **GitHub repository URL** fields described above. You can find these values under **Settings** > **Developer settings** > **GitHub Apps** > **[your app]** on GitHub.

Reviewer: ```suggestion
To register an existing GitHub App in Seqera:
1. Set the **GitHub credential type** to **GitHub App** and select **Add preexisting**
1. Enter the **GitHub URL**, **App scope**, and, if required, the **GitHub repository URL** described above.
1. Enter the app's security keys. To find these values, go to **Settings** > **Developer settings** > **GitHub Apps** > **[your app]** on GitHub:
    - App ID
    - Installation ID
    - App slug
    - Private key
    - Client secret
    - Webhook secret
1. Select **Add** to save the credentials.
```

#### Handling duplicate credentials

> Seqera enforces uniqueness of GitHub App credentials by **Repository URL** within the same workspace or user context. If you attempt to create a credential — through either the manifest flow or the existing-app flow — for a repository URL that already has a GitHub App credential, the operation fails with a duplicate error and no new credential is stored.

Reviewer: ```suggestion
Seqera enforces uniqueness of GitHub App credentials by **Repository URL** within the same workspace or user context. If a GitHub App credential already exists for a given repository URL, any attempt to create another (through either the manifest flow or the existing-app flow) fails with a duplicate error. No new credential is stored.
```

To resolve a duplicate:

- **Reuse the existing credential** — In most cases the existing credential already grants Seqera the access it needs. Open it from the **Credentials** list to confirm the installed app and repository association.
- **Delete the obsolete credential first** — If the existing credential is stale (for example, the app has been uninstalled or the private key was rotated outside of Seqera), delete it from the **Credentials** list and then re-run the creation flow.

> - **Use a different repository URL or leave the field empty** — If you need a second credential covering a broader scope, omit the **Repository URL** or use a different one. Seqera's [credential filtering](#multiple-credential-filtering) then selects the most specific match at launch time.

Reviewer: ```suggestion
- **Reuse the existing credential**: In most cases the existing credential already grants Seqera the access it needs. Open it from the **Credentials** list to confirm the association between the installed app and the repository.
- **Delete the obsolete credential first**: If the existing credential is stale (e.g., the app has been uninstalled or the private key was rotated outside of Seqera), delete it from the **Credentials** list and then re-run the creation flow.
- **Use a different repository URL or leave the field empty**: If you need a second credential covering a broader scope, omit the **Repository URL** or use a different one. Seqera's [credential filtering](#multiple-credential-filtering) then selects the most specific match at launch time.
```
